App.js

```diff
@@ -3,15 +3,52 @@ import {StyleSheet, Text, View} from 'react-native';
 
 import Header from './header';
 import Footer from './footer';
+import { PRIORITIES } from './constants';
+
+// type Props = {};
+export default class App extends Component { // <Props> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      text: "",
+      priority: PRIORITIES.MEDIUM,
+      items: []
+    }
+    this.handleAddItem = this.handleAddItem.bind(this);
+  }
+
+  handleAddItem() {
+    if (!this.state.text) return; // ignore empty strings
+    const newItems = [
+      ...this.state.items,
+      {
+        key: Date.now(),
+        text: this.state.text,
+        priority: this.state.priority,
+        complete: false
+      }
+    ]
+    this.setState({
+      items: newItems,
+      text: "",
+      priority: PRIORITIES.MEDIUM
+    })
+  }
 
-type Props = {};
-export default class App extends Component<Props> {
   render() {
     return (
       <View style={styles.container}>
-        <Header />
+        <Header
+          value={this.state.text}
```

Comment:
Mislim da bi text i priority trebalo da budu u state-u same komponente, posto to jeste njen state, ne state parent komponente. Separation of concerns